src/utils/gpu_arrays_functions.h

```diff
@@ -1,10 +1,56 @@
 #ifndef GPU_ARRAY_FUNCTIONS_H
 #define GPU_ARRAY_FUNCTIONS_H
 
-#include "../global/global.h"
+#include "../utils/error_handling.h"
+#include "../utils/gpu.hpp"
+#include "../global/global_cuda.h"
+#include "../utils/gpu_arrays_functions.h"
+#include <iostream>
 
-void Extend_GPU_Array_Real( Real **current_array_d, int current_size, int new_size, bool print_out );
 
+template< typename T > void Extend_GPU_Array( T **current_array_d, int current_size, int new_size, bool print_out ){
```

Comment:
That makes sense, I didn’t know it already existed. 

It’s not the reason I requested a change. I want to make sure the new CUDA rand stuff support setting the seed so we can have reliable tests; requesting a change was the only way to do that since the other options are “comment” and “approve” which don’t seem suitable to address a potential regression.

Reply:
Agreed! I also assumed as much, but wanted to clarify since it kind of appears in a bundle on github.